### PR TITLE
Updating kubeadm docs with KDD info

### LINKED ISSUE #963: Make KDD manifest work with kubeadm
The KDD manifests work with kubeadm, but currently require changing the default IP Pool to match the one kubeadm sets (or vice-versa).

We document it, but its [hidden away on the KDD page](http://docs.projectcalico.org/v2.3/getting-started/kubernetes/installation/hosted/kubernetes-datastore/).

Would be good to at [least point to it from the kubeadm doc](http://docs.projectcalico.org/v2.3/getting-started/kubernetes/installation/hosted/kubeadm/)

## Expected Behavior

KDD should work with kubeadm out-of-the-box, or we should at least document how to do it in the kubeadm doc.

## Current Behavior

Hard to find the _minimal_ docs for running KDD with kubeadm.

### PR DESCRIPTION
**Pull request by heschlie**
## Description

Fixes #963 

I've pulled info from the KDD page and fit it in here, so we can represent how to bring up a kubeadm cluster with Calico using either etcd or KDD.

I also tried to make it fairly clear at the start that the cluster will need to have been initialized with `--pod-network-cidr=192.168.0.0/16`.

I did not pull over anything about policy only, or user supplied networking, but can if we feel it belongs here.

## Todos

- [ ] Tests
- [x] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md [hunk 1]
```diff
@@ -2,41 +2,84 @@
...
 For Kubeadm 1.5 with Kubernetes version v1.5.x:
 
-```
+```shell
 kubectl apply -f {{site.url}}/{{page.version}}/getting-started/kubernetes/installation/hosted/kubeadm/1.5/calico.yaml
 ```
 
 >[Click here to view the above yaml directly.](1.5/calico.yaml)
 
+### Kubernetes Datastore
```
Comment (caseydavenport) [inline master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: I think I'd vote for compressing this section into something a bit more concise, without all the explanation for RBAC and BGP networking.

For a user coming here, they basically want "What commands can I run to get networking and policy"

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md [hunk 2]
```diff
@@ -2,41 +2,68 @@
 title: Kubeadm Hosted Install
 ---
 
-This document outlines how to install Calico, as well as a as single node
-etcd cluster for use by Calico on a Kubernetes cluster created by kubeadm.
+This document outlines how to install Calico on a kubeadm cluster.
```
Comment (emanic) [inline master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: Much better sentence!!!

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md [hunk 3]
```diff
@@ -2,41 +2,68 @@
...
 
-#### Installation
+The below manifests are compatible with kubeadm clusters initialized with a
+`pod-network-cidr` matching the default pool of `192.168.0.0/16`, as follows:
+
+```shell
+kubeadm init --pod-network-cidr=192.168.0.0/16
+```
+
+### ETCD Datastore
```
Comment (emanic) [inline master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: I am not sure about the all caps for etcd....I believe that it is always lowercased `etcd`. I am seeing it that way even when it begins a sentence, so we should stick to that convention as well. https://github.com/coreos/etcd Also, I am going to be recommending sentence case for headings, (to be discussed soon), so I would recommend a lowercase for datastore as well.

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md [hunk 4]
```diff
@@ -2,41 +2,68 @@
...
-```
+```shell
 kubectl apply -f {{site.url}}/{{page.version}}/getting-started/kubernetes/installation/hosted/kubeadm/1.5/calico.yaml
 ```
 
 >[Click here to view the above yaml directly.](1.5/calico.yaml)
 
+### Kubernetes Datastore
+
+To install Calico with Calico networking, run the following commands to create these necessary RBAC
```
Comment (emanic) [inline master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: This sentence is a little awkward....I would suggest going to a list to help out, so it would be something like:

"To install Calico with Calico networking, run the following commands to:
1. Create the necessary RBAC roles and bindings.
1. Launch Calico.

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md [hunk 5]
```diff
@@ -2,41 +2,68 @@
...
 ```
 
 >[Click here to view the above yaml directly.](1.5/calico.yaml)
 
+### Kubernetes Datastore
+
+To install Calico with Calico networking, run the following commands to create these necessary RBAC 
+roles and bindings, and launch Calico.
+
+> **Note**: The following RBAC policy is compatible with the Kubernetes v1.7+
```
Comment (emanic) [inline master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: "the Kubernetes v1.7+" --> "Kubernetes v1.7."  

Note the period at the end! :))

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md [hunk 6]
```diff
@@ -2,41 +2,68 @@
...
+
+> **Note**: Calico v2.5.0 or later with Kubernetes backend requires Kubernetes 
+> v1.7.0 or higher.
```
Comment (emanic) [inline master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: I think it would better to say "later" instead of "higher"

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md [hunk 7]
```diff
@@ -2,41 +2,68 @@
...
 If you have already built your cluster with kubeadm, please review the
 [Requirements / Limitations](#requirements--limitations) at the bottom of
 this page. It is likely you will need to recreate your cluster with the
 `--pod-network-cidr` and `--service-cidr` arguments to kubeadm.
 
-Users who have deployed their own etcd cluster outside of kubeadm should
-use the [Calico only manifest](../hosted) instead, as it does not deploy its
-own etcd.
-
 You can easily create a cluster compatible with this manifest by following [the official kubeadm guide](http://kubernetes.io/docs/getting-started-guides/kubeadm/).
```
Comment (caseydavenport) [inline master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: "this manifest" - there are now two manifests :)

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md (hunk 3 above)
Comment (caseydavenport) [reply master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: Yeah, it should be `etcd`

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md (hunk 4 above)
Comment (caseydavenport) [reply master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: I think you should drop the "with Calico networking" bit - it doesn't make sense in the context of this doc.

Maybe something like this works better?  @emanic can confirm.

"To install Calico using the Kubernetes API as its datastore..."

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md (hunk 6 above)
Comment (caseydavenport) [reply master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: It should also say "This manifest requires" rather than "Calico v2.5.0"

Comment (heschlie) [reply master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: Maybe just condense the two down to "The following manifests require Kubernetes 1.7.0 or later"?

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md (hunk 4 above)
Comment (emanic) [reply master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: I must say, as yet I do not understand what "Calico networking" means, so I love the idea of dropping it. If what you mean is the "kdd" option, then I would suggest "To install Calico to use the Kubernetes API server, run the following commands to:"

(Since the Kubernetes API is not a _datastore_, per se...)

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md (hunk 6 above)
Comment (emanic) [reply master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: Yes, I think that would work great.

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md (hunk 4 above)
Comment (caseydavenport) [reply master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: Yep.  I think we need to say something instead of data store, if we're not comfortable with that terminology. "To use the kubernetes API server" is a bit vague and technically also applies to Calico etcd mode.

So maybe:

"To install Calico configured to use the Kubernetes API as its data source, run the following commands..."

Comment (emanic) [reply master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: The sentence gets a little unwieldy as we add in more about the "kdd" option, so perhaps we should break it out. Do we have an equivalent manifest for Calico with etcd? If so, we could present both options.

Comment (caseydavenport) [reply master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: The equivalent manifest for etcd is discussed up above in this doc.

**Comment by heschlie** [comment]
Thanks @caseydavenport and @emanic , I've addressed the feedback so far, let me know if we still need some tweaking.

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md [hunk 8]
```diff
@@ -2,41 +2,64 @@
...
 
-Users who have deployed their own etcd cluster outside of kubeadm should
-use the [Calico only manifest](../hosted) instead, as it does not deploy its
-own etcd.
+You can easily create a cluster compatible with these manifests by following [the official kubeadm guide](http://kubernetes.io/docs/getting-started-guides/kubeadm/).
 
-You can easily create a cluster compatible with this manifest by following [the official kubeadm guide](http://kubernetes.io/docs/getting-started-guides/kubeadm/).
+## Installation
 
+The below manifests are compatible with kubeadm clusters initialized with a
```
Comment (caseydavenport) [inline master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: I'd suggest moving the line above ("You can easily create . . .") down to be the first sentence in this section, since it talks about installing a kubeadm cluster.

Would also suggest rewording of this sentence so that it doesn't sound so much like there's a limitation in Calico that the 192.168 cidr be used.  Maybe something like this?

"Note that `--pod-network-cidr` is not always required when creating a kubeadm cluster.  When it is specified, the value must match the IPV4_POOL_CIDR in the Calico manifest.  Calico uses `192.168.0.0/16` by default.

For example:

```
kubeadm init --pod-network-cidr=192.168.0.0/16
```
"

^ That wording isn't great but hopefully shows the direction I'm thinking we should go.

Comment (heschlie) [reply master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: We require the `--pod-network-cidr` to be set for both etcd and kdd correct? If not then I should probably also move that statement to the Kubernetes section.

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md [hunk 9]
```diff
@@ -2,41 +2,64 @@
 title: Kubeadm Hosted Install
```
Comment (emanic) [inline master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: I am seeing kubeadm lowercased even when it begins a sentence in the k8s docs so we should probably do the same, as well. https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md (hunk 8 above)
Comment (caseydavenport) [reply master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: > Note that --pod-network-cidr is not always required when creating a kubeadm cluster

Other than that, yeah the same statement applies to both manifests.

**Comment by heschlie** [comment]
@caseydavenport @emanic I've reworded the section about `--pod-network-cidr` and I think it's getting the point across better, but it might still need improvement, WDYT?

**Comment by emanic** [comment]
I would just ask if you could squash your commits, that would be great.

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md [hunk 10]
```diff
@@ -1,45 +1,77 @@
...
+
+Calico installations require kubeadm initialized clusters to have set a `--pod-network-cidr`
+which should match the value set by `CALICO_IPV4POOL_CIDR` in the manifest:
+
+```yaml
+- name: CALICO_IPV4POOL_CIDR
+  value: "192.168.0.0/16"
+```
+
+This value can be changed to any CIDR that does not overlap the clusters service CIDR.
```
Comment (caseydavenport) [inline master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: typo: `cluster's`

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md [hunk 11]
```diff
@@ -1,45 +1,77 @@
...
 
-```
+```shell
 kubectl apply -f {{site.url}}/{{page.version}}/getting-started/kubernetes/installation/hosted/kubeadm/1.6/calico.yaml
 ```
 
 >[Click here to view the above yaml directly.](1.6/calico.yaml)
 
-For Kubeadm 1.5 with Kubernetes version v1.5.x:
+For kubeadm 1.5 with Kubernetes version v1.5.x:
```
Comment (caseydavenport) [inline master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: strictly speaking we can remove this bit - we don't need to support v1.5 in our docs any more.

Comment (caseydavenport) [reply master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: happy for that be a new PR though.

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md [hunk 12]
```diff
@@ -64,8 +88,8 @@ $ kubectl get node <master_name> -o yaml
 * This install does not configure etcd TLS
 * This install expects that one Kubernetes master node has been labeled
```
Comment (caseydavenport) [inline master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: This only applies to the etcd manifest.  I wonder if this is really useful information or if it can be dropped - I would think maybe we just drop this line.

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md [hunk 13]
```diff
@@ -1,45 +1,69 @@
...
+>[Click here to view the above Calico yaml directly.](../kubernetes-datastore/calico-networking/1.7/calico.yaml)
 
-## Using calicoctl in a Kubeadm Cluster
+## Using calicoctl in a kubeadm cluster
 
-The simplest way to use calicoctl in Kubeadm is by running it as a pod.
+The simplest way to use calicoctl in kubeadm is by running it as a pod.
 See [using calicoctl with Kubernetes](../../../tutorials/using-calicoctl#b-running-calicoctl-as-a-kubernetes-pod) for more information.
 
 ## About
```
Comment (caseydavenport) [inline master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: This about section doesn't make sense any more given the changes in this document - it only applies to etcd but is scoped such that it looks like it applies to the kdd section as well.

I wonder if it's useful any more - I would probably vote to remove it entirely, perhaps splitting the information into other sections where it makes sense.

Comment (heschlie) [reply master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: It definitely feels more at home in the etcd section, the first two paragraphs seem helpfule, but I'm not sure how helpful it is to tell them how to check for the label, but it might be helpful to include what that label is. So I was thinking, drop the blurb about checking for the label, but specifying what that label is in the paragraph about how we pick where to run the etcd pod.

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md [hunk 14]
```diff
@@ -1,45 +1,69 @@
...
+```
+
+### etcd datastore
+
+Users who have deployed their own etcd cluster outside of kubeadm should
+use the [Calico only manifest](../hosted) instead, as it does not deploy its
+own etcd.
 
 To install this Calico and a single node etcd on a run the following command
 depending on your kubeadm / kubernetes version:
```
Comment (heschlie) [inline master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: This doesn't make much sense now that we only have a v1.6.0+ manifest.

File: master/getting-started/kubernetes/installation/hosted/kubeadm/index.md (hunk 13 above)
Comment (caseydavenport) [reply master/getting-started/kubernetes/installation/hosted/kubeadm/index.md]: I think at this point just remove the rest here - we don't need the checking for the label bit.

**Comment by bcreane** [comment]
The build failure is a side-effect of htmlproofer not handling partially published releases - the doc references v2.6 github objects that don't exist yet. For now, this will require that a maintainer override the UTs and accept the PR.

**Comment by heschlie** [comment]
I've trimmed out the about section and trimmed down the requirements section, and brought this up to date with master to resolve the conflicts, I think we're good for another round.
